### ISSUE TITLE
simple bug and am close (hopefully 😁) to making PR: login command hangs if profile doesn't exist

### ISSUE THREAD

**Issue by matthewliuswims** [99designs/aws-vault#545]
- [x] I am using the latest release of AWS Vault
- [x] I have provided my `.aws/config` (redacted if necessary)
- [x] I have provided the debug output using `aws-vault --debug` (redacted if necessary)

# bug/replication
`aws-vault login <non-existent-profile>` hangs forever if profile doesn't exist 😓

# fix attempt
I did some digging: the hang is caused by infinite recursion here https://github.com/99designs/aws-vault/blob/1ce365597489943c2894d3265280933af28de102/vault/vault.go#L220-L231 

```
hasMasterCreds, err := keyring.Has(profileName)  // the culprit
// hasMasterCreds, err end up being false and nil respectively forever and ever
```

func is called here 
https://github.com/99designs/aws-vault/blob/1ce365597489943c2894d3265280933af28de102/vault/vault.go#L196-L197

I'm happy to make a pr, but i'm not sure what kind of base case checking I should be doing for `MasterCredentialsFor` func without breaking functionality 

Thank you!

**Comment by mtibben** [comment]
A PR would certainly be welcome! Get something going and we can discuss

**Comment by mtibben** [comment]
How's that PR coming along @matthewliuswims? :)

**Comment by mtibben** [comment]
Possible solution here? https://github.com/99designs/aws-vault/issues/507#issuecomment-577427034

**Comment by matthewliuswims** [comment]
ah yes - that looks like a great fix. :) gonna do some testing and i'll hopefully put in a PR today!

**Comment by matthewliuswims** [comment]
@mtibben PR here https://github.com/99designs/aws-vault/pull/575/files !

**Comment by mtibben** [comment]
Fixed in 1496990
